### PR TITLE
Remove LinearRegressor from default AutoMLSearch estimators

### LINKED ISSUE #2589: regress.csv breaking since 0.29.0 release
Looking glass generates this for the regress.csv dataset.

```
Downloading dataset regress.csv with target y from bucket evalml-looking-glass-datasets
--
Running dataset regress.csv trial 1 / 5
Generating pipelines to search over...
8 pipelines ready for search.
*****************************
* Beginning pipeline search *
*****************************
Optimizing for R2.
Greater score is better.
Using SequentialEngine to train and score pipelines.
Searching up to 1 batches for a total of 9 pipelines.
Allowed model families: lightgbm, decision_tree, random_forest, catboost, xgboost, extra_trees, linear_model
Evaluating Baseline Pipeline: Mean Baseline Regression Pipeline
Mean Baseline Regression Pipeline:
Starting cross validation
Finished cross validation - mean R2: -0.002
High coefficient of variation (cv >= 0.2) within cross validation scores.
Mean Baseline Regression Pipeline may not perform as estimated on unseen data.
*****************************
* Evaluating Batch Number 1 *
*****************************
AutoML search raised a fatal exception: R2 encountered ValueError with message (y_predicted contains NaN or infinity):
File "/evalml_looking_glass/evalml/evalml/pipelines/pipeline_base.py", line 339, in _score_all_objectives
score = self._score(
File "/evalml_looking_glass/evalml/evalml/pipelines/pipeline_base.py", line 310, in _score
return objective.score(y, predictions, X)
File "/evalml_looking_glass/evalml/evalml/objectives/objective_base.py", line 81, in score
self.validate_inputs(y_true, y_predicted)
File "/evalml_looking_glass/evalml/evalml/objectives/objective_base.py", line 130, in validate_inputs
raise ValueError("y_predicted contains NaN or infinity")
ExpVariance encountered ValueError with message (y_predicted contains NaN or infinity):
File "/evalml_looking_glass/evalml/evalml/pipelines/pipeline_base.py", line 339, in _score_all_objectives
score = self._score(
File "/evalml_looking_glass/evalml/evalml/pipelines/pipeline_base.py", line 310, in _score
return objective.score(y, predictions, X)
File "/evalml_looking_glass/evalml/evalml/objectives/objective_base.py", line 81, in score
self.validate_inputs(y_true, y_predicted)
File "/evalml_looking_glass/evalml/evalml/objectives/objective_base.py", line 130, in validate_inputs
raise ValueError("y_predicted contains NaN or infinity")
MaxError encountered ValueError with message (y_predicted contains NaN or infinity):
File "/evalml_looking_glass/evalml/evalml/pipelines/pipeline_base.py", line 339, in _score_all_objectives
score = self._score(
File "/evalml_looking_glass/evalml/evalml/pipelines/pipeline_base.py", line 310, in _score
return objective.score(y, predictions, X)
File "/evalml_looking_glass/evalml/evalml/objectives/objective_base.py", line 81, in score
self.validate_inputs(y_true, y_predicted)
File "/evalml_looking_glass/evalml/evalml/objectives/objective_base.py", line 130, in validate_inputs
raise ValueError("y_predicted contains NaN or infinity")
MedianAE encountered ValueError with message (y_predicted contains NaN or infinity):
File "/evalml_looking_glass/evalml/evalml/pipelines/pipeline_base.py", line 339, in _score_all_objectives
score = self._score(
File "/evalml_looking_glass/evalml/evalml/pipelines/pipeline_base.py", line 310, in _score
return objective.score(y, predictions, X)
File "/evalml_looking_glass/evalml/evalml/objectives/objective_base.py", line 81, in score
self.validate_inputs(y_true, y_predicted)
File "/evalml_looking_glass/evalml/evalml/objectives/objective_base.py", line 130, in validate_inputs
raise ValueError("y_predicted contains NaN or infinity")
MSE encountered ValueError with message (y_predicted contains NaN or infinity):
File "/evalml_looking_glass/evalml/evalml/pipelines/pipeline_base.py", line 339, in _score_all_objectives
score = self._score(
File "/evalml_looking_glass/evalml/evalml/pipelines/pipeline_base.py", line 310, in _score
return objective.score(y, predictions, X)
File "/evalml_looking_glass/evalml/evalml/objectives/objective_base.py", line 81, in score
self.validate_inputs(y_true, y_predicted)
File "/evalml_looking_glass/evalml/evalml/objectives/objective_base.py", line 130, in validate_inputs
raise ValueError("y_predicted contains NaN or infinity")
MAE encountered ValueError with message (y_predicted contains NaN or infinity):
File "/evalml_looking_glass/evalml/evalml/pipelines/pipeline_base.py", line 339, in _score_all_objectives
score = self._score(
File "/evalml_looking_glass/evalml/evalml/pipelines/pipeline_base.py", line 310, in _score
return objective.score(y, predictions, X)
File "/evalml_looking_glass/evalml/evalml/objectives/objective_base.py", line 81, in score
self.validate_inputs(y_true, y_predicted)
File "/evalml_looking_glass/evalml/evalml/objectives/objective_base.py", line 130, in validate_inputs
raise ValueError("y_predicted contains NaN or infinity")
Root Mean Squared Error encountered ValueError with message (y_predicted contains NaN or infinity):
File "/evalml_looking_glass/evalml/evalml/pipelines/pipeline_base.py", line 339, in _score_all_objectives
score = self._score(
File "/evalml_looking_glass/evalml/evalml/pipelines/pipeline_base.py", line 310, in _score
return objective.score(y, predictions, X)
File "/evalml_looking_glass/evalml/evalml/objectives/objective_base.py", line 81, in score
self.validate_inputs(y_true, y_predicted)
File "/evalml_looking_glass/evalml/evalml/objectives/objective_base.py", line 130, in validate_inputs
raise ValueError("y_predicted contains NaN or infinity")
File "/evalml_looking_glass/evalml/evalml/automl/engine/engine_base.py", line 217, in train_and_score_pipeline
scores = cv_pipeline.score(X_valid, y_valid, objectives=objectives_to_score)
File "/evalml_looking_glass/evalml/evalml/pipelines/regression_pipeline.py", line 55, in score
return self._score_all_objectives(
File "/evalml_looking_glass/evalml/evalml/pipelines/pipeline_base.py", line 351, in _score_all_objectives
raise PipelineScoreError(exceptions, scored_successfully)
Encountered error on dataset regress.csv trial 1 / 5:
<class 'evalml.exceptions.exceptions.PipelineScoreError'>: R2 encountered ValueError with message (y_predicted contains NaN or infinity):
File "/evalml_looking_glass/evalml/evalml/pipelines/pipeline_base.py", line 339, in _score_all_objectives
score = self._score(
File "/evalml_looking_glass/evalml/evalml/pipelines/pipeline_base.py", line 310, in _score
return objective.score(y, predictions, X)
File "/evalml_looking_glass/evalml/evalml/objectives/objective_base.py", line 81, in score
self.validate_inputs(y_true, y_predicted)
File "/evalml_looking_glass/evalml/evalml/objectives/objective_base.py", line 130, in validate_inputs
raise ValueError("y_predicted contains NaN or infinity")
ExpVariance encountered ValueError with message (y_predicted contains NaN or infinity):
File "/evalml_looking_glass/evalml/evalml/pipelines/pipeline_base.py", line 339, in _score_all_objectives
score = self._score(
File "/evalml_looking_glass/evalml/evalml/pipelines/pipeline_base.py", line 310, in _score
return objective.score(y, predictions, X)
File "/evalml_looking_glass/evalml/evalml/objectives/objective_base.py", line 81, in score
self.validate_inputs(y_true, y_predicted)
File "/evalml_looking_glass/evalml/evalml/objectives/objective_base.py", line 130, in validate_inputs
raise ValueError("y_predicted contains NaN or infinity")
MaxError encountered ValueError with message (y_predicted contains NaN or infinity):
File "/evalml_looking_glass/evalml/evalml/pipelines/pipeline_base.py", line 339, in _score_all_objectives
score = self._score(
File "/evalml_looking_glass/evalml/evalml/pipelines/pipeline_base.py", line 310, in _score
return objective.score(y, predictions, X)
File "/evalml_looking_glass/evalml/evalml/objectives/objective_base.py", line 81, in score
self.validate_inputs(y_true, y_predicted)
File "/evalml_looking_glass/evalml/evalml/objectives/objective_base.py", line 130, in validate_inputs
raise ValueError("y_predicted contains NaN or infinity")
MedianAE encountered ValueError with message (y_predicted contains NaN or infinity):
File "/evalml_looking_glass/evalml/evalml/pipelines/pipeline_base.py", line 339, in _score_all_objectives
score = self._score(
File "/evalml_looking_glass/evalml/evalml/pipelines/pipeline_base.py", line 310, in _score
return objective.score(y, predictions, X)
File "/evalml_looking_glass/evalml/evalml/objectives/objective_base.py", line 81, in score
self.validate_inputs(y_true, y_predicted)
File "/evalml_looking_glass/evalml/evalml/objectives/objective_base.py", line 130, in validate_inputs
raise ValueError("y_predicted contains NaN or infinity")
MSE encountered ValueError with message (y_predicted contains NaN or infinity):
File "/evalml_looking_glass/evalml/evalml/pipelines/pipeline_base.py", line 339, in _score_all_objectives
score = self._score(
File "/evalml_looking_glass/evalml/evalml/pipelines/pipeline_base.py", line 310, in _score
return objective.score(y, predictions, X)
File "/evalml_looking_glass/evalml/evalml/objectives/objective_base.py", line 81, in score
self.validate_inputs(y_true, y_predicted)
File "/evalml_looking_glass/evalml/evalml/objectives/objective_base.py", line 130, in validate_inputs
raise ValueError("y_predicted contains NaN or infinity")
MAE encountered ValueError with message (y_predicted contains NaN or infinity):
File "/evalml_looking_glass/evalml/evalml/pipelines/pipeline_base.py", line 339, in _score_all_objectives
score = self._score(
File "/evalml_looking_glass/evalml/evalml/pipelines/pipeline_base.py", line 310, in _score
return objective.score(y, predictions, X)
File "/evalml_looking_glass/evalml/evalml/objectives/objective_base.py", line 81, in score
self.validate_inputs(y_true, y_predicted)
File "/evalml_looking_glass/evalml/evalml/objectives/objective_base.py", line 130, in validate_inputs
raise ValueError("y_predicted contains NaN or infinity")
Root Mean Squared Error encountered ValueError with message (y_predicted contains NaN or infinity):
File "/evalml_looking_glass/evalml/evalml/pipelines/pipeline_base.py", line 339, in _score_all_objectives
score = self._score(
File "/evalml_looking_glass/evalml/evalml/pipelines/pipeline_base.py", line 310, in _score
return objective.score(y, predictions, X)
File "/evalml_looking_glass/evalml/evalml/objectives/objective_base.py", line 81, in score
self.validate_inputs(y_true, y_predicted)
File "/evalml_looking_glass/evalml/evalml/objectives/objective_base.py", line 130, in validate_inputs
raise ValueError("y_predicted contains NaN or infinity")
File "looking_glass/container/perf_test_runner.py", line 232, in run
trial_measurements_dict = measure_evalml_automl(X, y,
File "/evalml_looking_glass/looking_glass/container/evalml_runner.py", line 65, in measure_evalml_automl
automl.search(**evalml_search_args)
File "/evalml_looking_glass/evalml/evalml/automl/automl_search.py", line 840, in search
evaluation = computation.get_result()
File "/evalml_looking_glass/evalml/evalml/automl/engine/sequential_engine.py", line 39, in get_result
return self.work(**self.kwargs)
File "/evalml_looking_glass/evalml/evalml/automl/engine/engine_base.py", line 305, in evaluate_pipeline
return train_and_score_pipeline(
File "/evalml_looking_glass/evalml/evalml/automl/engine/engine_base.py", line 224, in train_and_score_pipeline
automl_config.error_callback(
File "/evalml_looking_glass/evalml/evalml/automl/callbacks.py", line 15, in raise_error_callback
raise exception
File "/evalml_looking_glass/evalml/evalml/automl/engine/engine_base.py", line 217, in train_and_score_pipeline
scores = cv_pipeline.score(X_valid, y_valid, objectives=objectives_to_score)
File "/evalml_looking_glass/evalml/evalml/pipelines/regression_pipeline.py", line 55, in score
return self._score_all_objectives(
File "/evalml_looking_glass/evalml/evalml/pipelines/pipeline_base.py", line 351, in _score_all_objectives
raise PipelineScoreError(exceptions, scored_successfully)
Uploading results for dataset regress.csv trial 1 / 5
Uploading artifacts for dataset regress.csv trial 1 / 5
Completed dataset regress.csv trial 1 / 5

```

It seems that this is not a product of the LogTransformer, but is a symptom of the LinearRegressor generating extremely large positive and negative predictions on certain rows.  

Here's a snippet of the regress.csv that generated large positive predictions: 
[hail_mary.csv](https://github.com/alteryx/evalml/files/6934528/hail_mary.csv)

### PR DESCRIPTION
**Pull request by freddyaboulton**
### Pull Request Description
Fixes #2589 

Rationale [here](https://alteryx.atlassian.net/wiki/spaces/PS/pages/986153264/Investigating+Looking+Glass+Failures+on+regress.csv?focusedCommentId=987758749)

-----
*After creating the pull request: in order to pass the **release_notes_updated** check you will need to update the "Future Release" section of* `docs/source/release_notes.rst` *to include this pull request by adding :pr:`123`.*

### CODE REVIEW

File: evalml/automl/automl_algorithm/evalml_algorithm.py [hunk 1]
```diff
@@ -129,7 +129,7 @@ def __init__(
     def _naive_estimators(self):
         if is_regression(self.problem_type):
             naive_estimators = [
-                "Linear Regressor",
+                "Elastic Net Regressor",
```
Comment (dsherry) [inline evalml/automl/automl_algorithm/evalml_algorithm.py:132]: Hm so were we not running elastic net in automl before this point? That doesn't match with my memory but perhaps I've just forgotten the details of the plumbing here, lol

Comment (freddyaboulton) [reply evalml/automl/automl_algorithm/evalml_algorithm.py:132]: This is on the new algorithm @jeremyliweishih wrote so not what's run by default in AutoMLSearch right now. In AutoMLSearch we run elastic net by default!

**Review by dsherry — APPROVED** [review]
🚢 🙏

**Review by bchen1116 — APPROVED** [review]
LGTM!

**Pull request merged**